xbmc/utils/FileUtils.cpp

```diff
@@ -226,3 +229,48 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   }
   return dateAdded;
 }
+
+bool CFileUtils::ZebraListAccessCheck(const std::string &filePath)
+{
+  // white/black list access checks, disallow exploits
+
+  // no access to the passwords.xml file,
+  // this can expose user/pass of remote servers
+  if (filePath.find("passwords.xml") != std::string::npos)
+  {
+    CLog::Log(LOGDEBUG,"http access denied");
+    return false;
+  }
+
+  // no access to the sources.xml file,
+  // this can expose user/pass of remote servers
+  if (filePath.find("sources.xml") != std::string::npos)
+  {
+    CLog::Log(LOGDEBUG,"http access denied");
+    return false;
+  }
+
+#if defined(TARGET_DARWIN)
+  char *fullpath = realpath(filePath.c_str(), nullptr);
+  if (fullpath)
+  {
+    const std::string testpath = fullpath;
+    free(fullpath);
+
+    // if this is a real path and accesses into user home, allow.
+    std::string userHome = CDarwinUtils::GetUserHomeDirectory();
```

Comment:
realpath is the key here, it will collapse any usage of "../../../" and hex encodings into a realpath.